### PR TITLE
Use Tuple only as return value in Bootstrap

### PR DESCRIPTION
**Pull request by rmuir**
I wanted to raise this separately (it impacts more of the code than just here). I think in general this Tuple should just be used only as a return-value mechanism and then unboxed by calling code. Otherwise we are stuck working with "variable" names v1 and v2, which can make code hard to follow: you only have types and no context about the instances.

### CODE REVIEW

**Comment by kimchy** [comment]
LGTM, ++

**Comment by mikemccand** [comment]
LGTM
